Exits with code 1 when using `--async`

### Current behavior
When running with `--async` flag, `saucectl run` exits with code 1.

### Desired behavior
If there's no errors, it should exit with code 0.

### Config to reproduce
Eg command:
```
saucectl run --async --select-suite ${{ github.event.inputs.sauce_suite }}
```

### Versions
Using Github Action `saucelabs/saucectl-run-action@v1` config set to `saucectl-version: v0.99.3`:

```
jobs:
  test-on-sauce:
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - uses: saucelabs/saucectl-run-action@v1
        with:
          saucectl-version: v0.99.3
          testing-environment: sauce
          skip-run: true

      - name: saucectl run
        run: |
          saucectl configure -u $SAUCE_USERNAME -a $SAUCE_ACCESS_KEY
          saucectl run --async --select-suite your-suite-name
```

The `saucectl run` step fails the job with `Error: Process completed with exit code 1.`

--- a/internal/saucecloud/cloud.go
+++ b/internal/saucecloud/cloud.go
@@ -119,7 +119,9 @@ func (r *CloudRunner) collectResults(artifactCfg config.ArtifactDownload, result
 	for i := 0; i < expected; i++ {
 		res := <-results
 		// in case one of test suites not passed
-		if !res.job.Passed {
+		// ignore jobs that are still in progress (i.e. async execution or client timeout)
+		// since their status is unknown
+		if job.Done(res.job.Status) && !res.job.Passed {
 			passed = false
 		}
 		completed++

